torchtune/models/clip/_model_builders.py

```diff
@@ -1,4 +1,63 @@
-from torchtune.models.clip._transforms import CLIPImageTransform
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+from pathlib import Path
+
+from torchtune.models.clip._text_encoder import CLIPTextEncoder
+from torchtune.models.clip._tokenizer import CLIPTokenizer
+from torchtune.models.clip._transform import CLIPImageTransform
+from torchtune.utils._download import TORCHTUNE_LOCAL_CACHE_FOLDER, download_file
+
+CLIP_VOCAB_URL = "https://github.com/openai/CLIP/raw/refs/heads/main/clip/bpe_simple_vocab_16e6.txt.gz"
+
+
+def clip_tokenizer(
+    vocab_path: Path = TORCHTUNE_LOCAL_CACHE_FOLDER / "clip_vocab.txt.gz",
+    download_if_missing: bool = True,
+    max_seq_len: int = 77,
+    truncate: bool = True,
+) -> CLIPTokenizer:
+    """
+    Builder for the CLIP text tokenizer.
+
+    Args:
+        vocab_path (pathlib.Path): Path to the CLIP vocab file
+            Default: '~/.cache/torchtune/clip_vocab.txt.gz'
+        download_if_missing (bool): Download the vocab file if it's not found
+            Default: True
+        max_seq_len (bool): Context length
+            Default: 77
+        truncate (bool): Truncate the token sequence if it exceeds max_seq_len (otherwise raises AssertionError)
+            Default: True
+
+    Returns:
+        CLIPTokenizer: Instantiation of the CLIP text tokenizer
+    """
+    if not vocab_path.exists():
+        assert download_if_missing, f"Missing CLIP tokenizer vocab: {vocab_path}"
+        download_file(CLIP_VOCAB_URL, vocab_path)
+
+    return CLIPTokenizer(vocab_path, max_seq_len=max_seq_len, truncate=truncate)
+
+
+def clip_text_encoder_large() -> CLIPTextEncoder:
```

Comment:
The name for this should be clip_text_vit_l14 (are there anymore qualifiers in the hf official name?)